jax/api.py

```diff
@@ -707,6 +715,25 @@ def pmap(fun, axis_name=None):
   [ 1.  1.  1.  1.]
   >>> print(doubly_normed.sum((0, 1)))
   1.0
+
+  The ``devices`` argument can be used to specify exactly which devices are used
```

Comment:
Not blocking for this PR, but we could add a description of how the order matters, i.e. that it is significant and corresponds to the order along the mapped axis.

Could also specify what happens on nesting (error, ideally?).